core/frontend/src/render/webgl/Technique.ts

```diff
@@ -192,15 +185,27 @@ export abstract class VariedTechnique implements Technique {
     assert(this._clippingPrograms[index] !== undefined);
   }
 
+  protected addLogDepth(builder: ProgramBuilder, flags: TechniqueFlags): void {
```

Comment:
make sure compiling-all-shaders test still succeeds

Reply:
I reverted the `VariedTechnique` changes and made `SkySphereTechnique` inherit from a new `Technique` implementation I called `MultipleTechnique`, which is basically `VariedTechnique` with none of the assumptions about clipping and eyeSpace I was having problems with.

I hit another snag around using a mat3 varying in the Atmosphere shader, [because mat3 is flagged as an invalid varying type](https://github.com/iTwin/itwinjs-core/blob/b341b47ed8a0f642b61755092a76eaa124991b50/core/frontend/src/render/webgl/ShaderBuilder.ts#L406).
I originally used a mat3 because there are two vec3 values, scattered light and reflected light, which are computed from the same underlying data, and separating them would be a big exercise.
I got around that by [computing the varyings from the mat3](https://github.com/iTwin/itwinjs-core/blob/42ce1b804f2733df064821e4e27c1f2f309e0703/core/frontend/src/render/webgl/glsl/Atmosphere.ts#L323-L333) instead of passing it through the varyings. This required a new VertexShaderComponent to be defined so that I can call these new functions.

After these two changes, all cover tests pass locally